### PR TITLE
chore: Update dark theme logos styling (Fixes #1209)

### LINKED ISSUE #1209: [BUG] Logo colors not appropriate for dark mode.
# GitHub and Algolia Logo blend into the dark background.

----

**Bug description**

When we switch to dark mode, the GitHub and Algolia logos retain their previous colours only. That is why, it makes it difficult to see them on the webpage.

**To Reproduce**

Visit the home page: https://react-tooltip.com/ and change the theme to dark mode.

**Expected behaviour**

GitHub and Algolia's logos should change to a lighter version when switched to dark mode.

**Screenshots**

image
image

### PR DESCRIPTION
**Pull request by rahulharpal1603**
This PR fixes the Issue #1209. Now, when you toggle dark mode on the page, the GitHub and Algolia logos adapt to the change in background colour:

Demo Video:  https://streamable.com/79bbls

Before:
image

After:
image

Before:
image

After:
image

## Summary by CodeRabbit

- **New Features**
	- Introduced conditional styling for SVG elements in the HomepageFeatures and HomepageSponsored components, allowing for dark mode adaptability.

- **Style**
	- Added new styles for SVGs in dark mode, improving contrast and visibility for a better user experience.

### CODE REVIEW

**Comment by rahulharpal1603** [comment]
@gabrieljablonski  Please review

File: docs/src/components/HomepageFeatures/index.tsx [hunk 1]
```diff
@@ -42,10 +42,15 @@ const FeatureList: FeatureItem[] = [
 ]
 
 function Feature({ title, Svg, description }: FeatureItem) {
+  let classVar = styles.featureSvg;
+  if(Svg === require('@site/static/img/github.svg').default)
+  {
+    classVar = styles.featureSvg +" "+ styles.githubLogo;
+  }
```
Comment (gabrieljablonski) [inline docs/src/components/HomepageFeatures/index.tsx:49]:
```suggestion
  let svgClassName = styles.featureSvg
  if (Svg === require('@site/static/img/github.svg').default) {
    svgClassName = `${styles.featureSvg} ${styles.githubLogo}`
  }

```

File: docs/src/components/HomepageFeatures/index.tsx [hunk 2]
```diff
@@ -42,10 +42,15 @@ const FeatureList: FeatureItem[] = [
...
   return (
     <div className={clsx('col col--4')}>
       <div className="text--center">
-        <Svg className={styles.featureSvg} role="img" />
+        <Svg className={classVar} role="img" />
```
Comment (gabrieljablonski) [inline docs/src/components/HomepageFeatures/index.tsx:53]:
```suggestion
        <Svg className={svgClassName} role="img" />
```

File: docs/src/components/HomepageFeatures/styles.module.css [hunk 3]
```diff
@@ -9,3 +9,7 @@
   height: 200px;
   width: 200px;
 }
+
+html[data-theme='dark'] .githubLogo {
+  fill: #fff;
+}
```
Comment (gabrieljablonski) [inline docs/src/components/HomepageFeatures/styles.module.css:15]:
```suggestion
}

```

File: docs/src/components/HomepageSponsored/index.tsx [hunk 4]
```diff
@@ -51,11 +51,16 @@ const SponsorList: SponsorItem[] = [
 ]
 
 function Feature({ title, Svg, link }: FeatureItem) {
+  let classVar = styles.featureSvg;
+  if(Svg === require('@site/static/img/Algolia-logo.svg').default)
+  {
+    classVar = styles.featureSvg +" "+ styles.AlgoliaLogo;
+  }
```
Comment (gabrieljablonski) [inline docs/src/components/HomepageSponsored/index.tsx:58]:
```suggestion
  let svgClassName = styles.featureSvg
  if (Svg === require('@site/static/img/Algolia-logo.svg').default) {
    svgClassName = `${styles.featureSvg} ${styles.AlgoliaLogo}`
  }

```

File: docs/src/components/HomepageSponsored/index.tsx [hunk 5]
```diff
@@ -51,11 +51,16 @@ const SponsorList: SponsorItem[] = [
...
   return (
     <div className={clsx('col col--6')}>
       <div className="text--center">
         <a href={link} title={title} target="_blank" rel="noreferrer" aria-label={title}>
-          <Svg className={styles.featureSvg} role="img" />
+          <Svg className={classVar} role="img"/>
```
Comment (gabrieljablonski) [inline docs/src/components/HomepageSponsored/index.tsx:63]:
```suggestion
          <Svg className={svgClassName} role="img"/>
```

File: docs/src/components/HomepageSponsored/styles.module.css [hunk 6]
```diff
@@ -29,3 +29,7 @@
     width: 100%;
   }
 }
+
+html[data-theme='dark'] .AlgoliaLogo{
+  fill: #fff;
+}
```
Comment (gabrieljablonski) [inline docs/src/components/HomepageSponsored/styles.module.css:35]:
```suggestion
}

```

File: docs/static/img/Algolia-logo.svg
Comment (gabrieljablonski) [inline docs/static/img/Algolia-logo.svg:1]: Could you please explain why changing the SVG was needed?

**Review by gabrieljablonski — CHANGES_REQUESTED** [review]
Looks mostly good, but please see the suggested changes.

They're mostly the result of running `yarn run prettier docs/src`, so the code style matches our eslint rules, but I've also renamed `classVar` to `svgClassName` for clarity.

File: docs/static/img/Algolia-logo.svg
Comment (rahulharpal1603) [reply docs/static/img/Algolia-logo.svg:1]: Yes, I change a code inside svg which controlled the fill colour of the logo. That part was preventing the change of fill colour using CSS property: "fill: #000"

So, I removed the following part from it:
<defs><style>.cls-1,.cls-2{fill:#000;}.cls-2{fill-rule:evenodd;}</style></defs>

**Review by gabrieljablonski — CHANGES_REQUESTED** [review]
Please try applying this patch for an alternative instead of checking for exactly the github and algolia logos.

https://gist.github.com/gabrieljablonski/ff6371cc273b17521f27b86630ead4aa

See if you have any suggestions.

**Comment by rahulharpal1603** [comment]
> Please try applying this patch for an alternative instead of checking for exactly the github and algolia logos.
> 
> https://gist.github.com/gabrieljablonski/ff6371cc273b17521f27b86630ead4aa
> 
> See if you have any suggestions.

Yes, that idea is better. I will make these changes and get back to you.

**Comment by rahulharpal1603** [comment]
@gabrieljablonski, I have now used an alternative approach that you suggested. Please review.

**Review by gabrieljablonski — APPROVED** [review]
LGTM, but please run `yarn run prettier docs/src/` as there are a few code-style issues.

Also waiting on @danielbarion to give a final word on this.

**Pull request merged**